### PR TITLE
[feature](Nereids): hashCode(), equals() to Plan

### PR DESCRIPTION
**Pull request by jackwener**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Add hashCode(), equals() for operator.

Add basic UT for them(need more detail test).

**future ticket**: add hashCode(), equals() and UT for `Expression`

## Checklist(Required)

1. Does it affect the original behavior: no
2. Has unit tests been added: no need
3. Has document been added or modified: no need
4. Does it need to update dependencies: no
5. Are there any changes that cannot be rolled back: no

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHeapSort.java [hunk 1]
```diff
@@ -71,6 +72,23 @@ public boolean hasLimit() {
...
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        PhysicalHeapSort that = (PhysicalHeapSort) o;
+        return offset == that.offset && Objects.equals(orderKeys, that.orderKeys);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHeapSort.java:84]: limit？

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalAggregate.java [hunk 2]
```diff
@@ -163,7 +163,7 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(groupByExpressionList, outputExpressionList, partitionExprList, aggPhase);
+        return Objects.hash(super.hashCode(), groupByExpressionList, outputExpressionList, partitionExprList, aggPhase);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalAggregate.java:166]: why need super ?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalSort.java [hunk 3]
```diff
@@ -84,6 +84,24 @@ public String toString() {
...
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        LogicalSort that = (LogicalSort) o;
+        return offset == that.offset && Objects.equals(orderKeys, that.orderKeys);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalSort.java:96]: logical sort has no limit, should we remove offset?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHeapSort.java [hunk 4]
```diff
@@ -71,6 +72,23 @@ public boolean hasLimit() {
...
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        PhysicalHeapSort that = (PhysicalHeapSort) o;
+        return offset == that.offset && limit == that.limit && Objects.equals(orderKeys, that.orderKeys);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHeapSort.java:84]: we need to discuss later, should we split PhysicalHeapSort to PhysicalHeapSort and PhysicalTopN

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalSort.java (hunk 3 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalSort.java:96]: we should remove all limit/offset from AbstractPlan/Plan, and provide a Limit plan

Comment (jackwener) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalSort.java:96]: Yes, But we can do it in another PR (support limit operator)

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**